### PR TITLE
Migrate Rector Laravel to community

### LINKED ISSUE #64: Moving Laravel Rector to the Laravel community :heart:
In the past, we've moved few Rector-framework packages from the core directly to their communities.
It gives the community full power and freedom to decide what to add to the upgrade sets :+1: 

See: 

* [getrector.org/blog/separating-typo3-and-nette-as-community-packages](https://getrector.org/blog/separating-typo3-and-nette-as-community-packages)
* https://github.com/rectorphp/rector-cakephp

Based on such a great experience and positive community feedback, we've decided to do this for a Laravel framework package, that deserves much more contributions and maintainer from their own community.

See tweet https://twitter.com/rectorphp/status/1585361016262639616 :)

<br>

At the moment, I'm speaking with @driftingly on possible takeover of this package :muscle: 

Thanks for reaching out and the @laravel community the spread the news :pray:

### PR DESCRIPTION
**Pull request by driftingly**
This PR updates the Rector Laravel namespace to complete the move from Rector to a community package.

Closes #64

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
+    "name": "driftingly/rector-laravel",
     "type": "rector-extension",
     "license": "MIT",
     "description": "Rector upgrades rules for Laravel Framework",
     "require": {
         "php": ">=8.1"
     },
     "require-dev": {
-        "rector/rector-src": "dev-main",
+        "rector/rector": "^0.14",
```
Comment (samsonasik) [inline composer.json:10]: you need to try use `dev-main` before next Rector release and ensure it keep working:

```suggestion
        "rector/rector": "dev-main",
```

**Comment by samsonasik** [comment]
You can check how to migrate in cakephp/upgrade PR for its migration https://github.com/cakephp/upgrade/pull/197 process:

- ensure use `"rector/rector": "dev-main"` and run unit tests to see if it keep working
- ensure cross dependencies possibly needed due to if it included in rector/rector, they are prefixed, eg: `webmozart/assert`, `symfony/string`, while using it here is not prefixed.

If you want to downgrade the source to use php < 8.1, eg: to support php 7.2, you may need to generate the `config/`, `src`, and `rules`, 'rules-tests` to use older version syntax, eg:

```php
<?php
// rector-downgrade.php
declare(strict_types=1);

use Rector\Config\RectorConfig;
use Rector\Set\ValueObject\DowngradeLevelSetList;

return static function (RectorConfig $rectorConfig): void {
    $rectorConfig->import(DowngradeLevelSetList::DOWN_TO_PHP_72);
};
```

and run once:

```
vendor/bin/rector process src tests stubs config/ --config rector-downgrade.php
```

Then, check the dependencies if it support php 7.2.

**Comment by driftingly** [comment]
Having trouble getting the tests to pass.  As an example `AnonymousMigrationsRectorTest` has `fixture.php.inc` with code: 

```php
<?php

namespace RectorLaravel\Tests\Rector\Class_\AnonymousMigrationsRector\Fixture;

use Illuminate\Database\Migrations\Migration;

class CreateUsersTable extends Migration
{
    // ...
}

?>
```

`BetterReflection` chokes on this.  Using a FQCN works for `BetterReflection`, but the rector rule doesn't pick it up so it fails there.

```php
<?php

namespace RectorLaravel\Tests\Rector\Class_\AnonymousMigrationsRector\Fixture;

class CreateUsersTable extends Illuminate\Database\Migrations\Migration
{
    // ...
}

?>
```

In total there are 4 failures caused by these `BetterReflection` errors. 

Any suggestions @zingimmick @samsonasik?

File: README.md [hunk 2]
```diff
@@ -4,20 +4,20 @@ See available [Laravel rules](/docs/rector_rules_overview.md)
...
 
-This package is already part of [rector/rector](http://github.com/rectorphp/rector) package, so it works out of the box.
+This package is a [Rector](https://github.com/rectorphp/rector) extension developed by the Laravel community.
 
-All you need to do is install the main package, and you're good to go:
+Install the `RectorLaravel` package as dependency:
 
 ```bash
-composer require rector/rector --dev
+driftingly/rector-laravel --dev
```
Comment (samsonasik) [inline README.md:12]: `composer require` still needed:

```suggestion
composer require riftingly/rector-laravel --dev
```

**Comment by samsonasik** [comment]
CI seems green, which part that error?

**Comment by zingimmick** [comment]
> ```
> class CreateUsersTable extends Illuminate\Database\Migrations\Migration
> ```

Should be:
```php
class CreateUsersTable extends \Illuminate\Database\Migrations\Migration
```

**Comment by zingimmick** [comment]
> CI seems green, which part that error?

@samsonasik Depending on the configuration of the [zend.assertions](https://www.php.net/manual/en/ini.core.php#ini.zend.assertions).

Error log:
```
assert($startLine > 0) in phar:///Users/zing/data/code/github/rector-laravel/vendor/phpstan/phpstan/phpstan.phar/vendor/ondrejmirtes/better-reflection/src/Reflection/ReflectionClass.php:180
 
 Caused by
 AssertionError: assert($startLine > 0)
 
 phar:///Users/zing/data/code/github/rector-laravel/vendor/phpstan/phpstan/phpstan.phar/vendor/ondrejmirtes/better-reflection/src/Reflection/ReflectionClass.php:180
 phar:///Users/zing/data/code/github/rector-laravel/vendor/phpstan/phpstan/phpstan.phar/vendor/ondrejmirtes/better-reflection/src/Reflection/ReflectionClass.php:251
 phar:///Users/zing/data/code/github/rector-laravel/vendor/phpstan/phpstan/phpstan.phar/src/Reflection/BetterReflection/BetterReflectionProvider.php:209
 phar:///Users/zing/data/code/github/rector-laravel/vendor/phpstan/phpstan/phpstan.phar/src/Reflection/ReflectionProvider/MemoizingReflectionProvider.php:59
 phar:///Users/zing/data/code/github/rector-laravel/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/NodeScopeResolver.php:2072
 phar:///Users/zing/data/code/github/rector-laravel/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/NodeScopeResolver.php:532
 phar:///Users/zing/data/code/github/rector-laravel/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/NodeScopeResolver.php:326
 /Users/zing/data/code/github/rector-laravel/vendor/rector/rector-src/packages/NodeTypeResolver/PHPStan/Scope/PHPStanNodeScopeResolver.php:333
 /Users/zing/data/code/github/rector-laravel/vendor/rector/rector-src/packages/NodeTypeResolver/PHPStan/Scope/PHPStanNodeScopeResolver.php:219
 /Users/zing/data/code/github/rector-laravel/vendor/rector/rector-src/src/Application/ChangedNodeScopeRefresher.php:95
 /Users/zing/data/code/github/rector-laravel/vendor/rector/rector-src/src/Rector/AbstractRector.php:370
 /Users/zing/data/code/github/rector-laravel/vendor/rector/rector-src/src/Rector/AbstractRector.php:254
 /Users/zing/data/code/github/rector-laravel/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:200
 /Users/zing/data/code/github/rector-laravel/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:114
 /Users/zing/data/code/github/rector-laravel/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:223
 /Users/zing/data/code/github/rector-laravel/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:91
 /Users/zing/data/code/github/rector-laravel/vendor/rector/rector-src/src/PhpParser/NodeTraverser/RectorNodeTraverser.php:33
 /Users/zing/data/code/github/rector-laravel/vendor/rector/rector-src/src/Application/FileProcessor.php:41
 /Users/zing/data/code/github/rector-laravel/vendor/rector/rector-src/src/Application/FileProcessor/PhpFileProcessor.php:105
 /Users/zing/data/code/github/rector-laravel/vendor/rector/rector-src/src/Application/FileProcessor/PhpFileProcessor.php:66
 /Users/zing/data/code/github/rector-laravel/vendor/rector/rector-src/src/Application/ApplicationFileProcessor.php:127
 /Users/zing/data/code/github/rector-laravel/vendor/rector/rector-src/packages/Testing/PHPUnit/AbstractRectorTestCase.php:204
 /Users/zing/data/code/github/rector-laravel/vendor/rector/rector-src/packages/Testing/PHPUnit/AbstractRectorTestCase.php:171
 /Users/zing/data/code/github/rector-laravel/vendor/rector/rector-src/packages/Testing/PHPUnit/AbstractRectorTestCase.php:119
 /Users/zing/data/code/github/rector-laravel/tests/Rector/Class_/AnonymousMigrationsRector/AnonymousMigrationsRectorTest.php:17
```

**Comment by samsonasik** [comment]
@driftingly @zingimmick that's need to be compatible with BetterReflection 6.x, I made a commit at https://github.com/driftingly/rector-laravel/pull/65/commits/45170081b278217f7ac4ecef36cf7cb05f22367e and https://github.com/driftingly/rector-laravel/pull/65/commits/5684e0508e15821df8ac4fb50ce68fefeae011ce for it.

**Comment by samsonasik** [comment]
@driftingly All green 🎉 , should be ok now, tested locally as well, feel free to take over :)

**Comment by driftingly** [comment]
Amazing! Thank you @samsonasik 🙌

**Comment by TomasVotruba** [comment]
Amazing work, thank you guys 👏🎉😊

**Comment by samsonasik** [comment]
Rector 0.14.7 tagged yesterday  https://github.com/rectorphp/rector/releases/tag/0.14.7

@driftingly you can now require "rector/rector": ^0.14.7" and make a release for this package :)

**Pull request merged**